packages/core/src/model.ts

```diff
@@ -0,0 +1,35 @@
+import { assign } from './actions';
+import { AssignAction, Assigner, EventObject, PropertyAssigner } from './types';
+import { mapValues } from './utils';
+
+export function createModel<TContext, TEvent extends EventObject>(
+  initialState: TContext
+): {
+  context: TContext;
+  withUpdaters: <
+    T extends {
+      [key: string]:
+        | Assigner<TContext, TEvent>
+        | PropertyAssigner<TContext, TEvent>;
+    }
+  >(
+    assigners: T
+  ) => {
+    context: TContext;
+    actions: {
+      [K in keyof T]: AssignAction<TContext, TEvent>;
+    };
+  };
+} {
+  return {
+    context: initialState,
+    withUpdaters: (assigners) => {
```

Comment:
q: why not a second parameter to the `createModel` itself?

Reply:
Impossible to infer the actions in TypeScript.